src/pudl/analysis/mcoe.py

```diff
@@ -417,6 +402,14 @@ def mcoe(
         all_gens (bool): if True, include attributes of all generators in the
             :ref:`generators_eia860` table, rather than just the generators
             which have records in the derived MCOE values. True by default.
+        all_gens_cols (bool): if True, include all attributes from generators in the
+            :ref:`generators_eia860` table, rather than just the attributes in
+            `default_gens_cols` and the list of extra_gens_cols.
+        extra_gens_cols (List): list of names of column attributes to
+            include from the :ref:`generators_eia860` table in addition to the
```

Comment:
If you've included type hints in the call signature, they aren't required in the docstring.